b2g/components/RemoteControlService.jsm

```diff
@@ -337,15 +367,7 @@ this.RemoteControlService = {
     SystemAppProxy._sendCustomEvent(REMOTE_CONTROL_EVENT, {action: "request-control-mode"});
 
     // Read already stored devices
-    try {
-      this._pairedDevices = JSON.parse(Services.prefs.getCharPref(REMOTECONTROL_PREF_DEVICES));
-    } catch(e) {
-      DEBUG && debug ("Parse stored devices error: " + e);
-
-      // Empty paired devices
-      this._pairedDevices = {};
-      Services.prefs.setCharPref(REMOTECONTROL_PREF_DEVICES, JSON.stringify(this._pairedDevices));
-    }
+    this._pairedDevices = this._pairingData.getPairedDevice();
```

Comment:
I'd like to write the empty device back to pref immediately after we discover an error.